http-client/Network/HTTP/Client/Manager.hs

```diff
@@ -212,9 +217,7 @@ reap baton wmapRef =
     loop = do
         threadDelay (5 * 1000 * 1000)
         mmapRef <- deRefWeak wmapRef
-        case mmapRef of
```

Comment:
I believe this form is less clear, and in particular the comment explaining what the `Nothing` case means is lost.